ddtrace/internal/writer.py

```diff
@@ -263,10 +258,8 @@ def __init__(
                 }
             )
 
-        self._encoder = Encoder(
-            max_size=self._buffer_size,
-            max_item_size=self._max_payload_size,
-        )
+        self._infer_endpoint_and_encoder(agent_url)
```

Comment:
Not sure if we want to do this here. This would be in the main thread and happens on import which would mean that we'd have potentially multiple blocking http requests on library import.

The question would be where we do this which gets difficult... the best place would probably be on writer start up in the thread or to try doing it before the first payload?

Reply:
Good point! AgentWriter start up would be good as if there's no agent detected there's probably no need to start the thread. WDYT?